edb/server/compiler/compiler.py

```diff
@@ -3682,15 +3693,41 @@ def _extract_extensions(
 def _extract_roles(
     global_schema: s_schema.Schema,
 ) -> immutables.Map[str, immutables.Map[str, Any]]:
-    roles = {}
-    for role in global_schema.get_objects(type=s_role.Role):
+    extracted_roles = {}
+    schema_roles = global_schema.get_objects(type=s_role.Role)
+    for role in schema_roles:
         role_name = str(role.get_name(global_schema))
-        roles[role_name] = immutables.Map(
+        extracted_roles[role_name] = dict(
             name=role_name,
             superuser=role.get_superuser(global_schema),
             password=role.get_password(global_schema),
         )
-    return immutables.Map(roles)
+
+    # To populate all_permissions, combine the permissions of each role
+    # and its ancestors.
+    role_memberships: MutableMapping[s_role.Role, list[s_role.Role]] = {}
+    role_permissions: MutableMapping[s_role.Role, Sequence[str]] = {}
+    for role in schema_roles:
+        role_memberships[role] = list(
+            role.get_bases(global_schema).objects(global_schema)
```

Comment:
Is this just direct parents, though?

Reply:
Lol but it was a one line fix so I really should have just included it in this PR